content/swiftui/concepts/viewmodifier/terms/dropDestination/dropDestination.md

```diff
@@ -0,0 +1,104 @@
+
+> **Note**: The data to be dragged and subsequently dropped must conform to the `Transferable` protocol. This ensures the data being dragged can be recognized and processed by other views or external applications. For more information see [protocols](https://www.codecademy.com/resources/docs/swift/protocols).
+
+## Syntax
+
+```pseudo
+View
+  .dropDestination(
+    for: MyDataType.self,
+    action: (droppedItems, location) -> Bool,
+    isTargeted: (isMyViewTargeted) -> Void
+  )
+```
+
+This method takes three required parameters:
+
+- `for:` The data type of the draggable item that will be dropped on this view.
+- `action:` A closure invoked when a draggable item is dropped on this view. It adds the dropped items to a designated array and stores the drop location (CGPoint) in a specified variable. Custom actions are defined within this closure. It returns true if the drop is successful and false otherwise.
+- `isTargeted:` A closure that is triggered when a draggable item hovers over this view, updating a boolean value to true when a draggable item is hovering over this view, and false when it is not.
+
+> **Note**: Both `action` and `isTargeted` are escaping closures. This means that they are executed after the function (they are passed to)  returns. For a drag-and-drop operation to work appropriately, the `.dropDestination()` method must return the value of the data dropped on the view before the `action` and `isTargeted` closures can be executed. For more information see [closures](https://www.codecademy.com/resources/docs/swift/closures).
+
+## Example
+
+```swift
+struct ContentView: View {
+  @State private var isDropTargeted = false
+  @State private var droppedPayload: String = "No text dropped yet"
+  @State private var myPayload: String = "I belong in a blue box."
+
+ var body: some View {
+   HStack(spacing: 50) {
+     Square(text: $myPayload, color: .pink)
+     .draggable(myPayload, preview: {
+       Text("Dragging...")
+       .padding()
+       .background(Color.purple)
+       .cornerRadius(25)
+     })
+   }
+  
+  Square(text: $droppedPayload, color: .blue)
+    .dropDestination(for: String.self) { receivedPayload, locationDrop in
+      if !receivedPayload.isEmpty {
+        droppedPayload = receivedPayload[0]
+        myPayload = " "
+        return true
+      }
+      return false
+      } isTargeted: {
+        isDropTargeted = $0
+    }
+    .brightness(isDropTargeted ? -0.2 : 0)
+    .scaleEffect(isDropTargeted ? 1.25 : 1)
+  }
+
+  struct Square: View {
+    @Binding var text: String
+    var color: Color
+    var body: some View {
+      ZStack {
+        color
+        Text(text)
+          .padding()
+          .bold()
+      }
+      .frame(width: 150, height: 150)
+      .cornerRadius(25)
+    }
+  }
```

Comment:
@KTom101 I added the definition for the Square custom view. I used the same custom view for the draggable term entry example #3602. Should I add the custom view definition to that example as well?

Reply:
@marjoriekohn Yes thank you. I left a comment there too.